src/lib/common/include/sol-update-modules.h

```diff
@@ -0,0 +1,88 @@
+/*
+ * This file is part of the Soletta Project
+ *
+ * Copyright (C) 2015 Intel Corporation. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in
+ *     the documentation and/or other materials provided with the
+ *     distribution.
+ *   * Neither the name of Intel Corporation nor the names of its
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#include <stdint.h>
+
+#include "sol-update.h"
+
+#include "sol-macros.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct sol_update {
+#ifndef SOL_NO_API_VERSION
+#define SOL_UPDATE_API_VERSION (1)
+    uint16_t api_version; /**< API version */
+#endif
+
+    struct sol_update_handle * (*check)(const char *url,
+        void (*cb)(void *data, int status,
+        const struct sol_update_info *response),
+        const void *data);
+
+    struct sol_update_handle * (*fetch)(const struct sol_update_info *response,
+        void (*cb)(void *data, int status, const char *file_path),
```

Comment:
Ok. So we would have something like: `sol_update_fetch` giving a `struct sol_update *` to feed `sol_update_install`, a la `struct sol_update_info *` which goes from `sol_update_check` to `sol_update_fetch` - but without the `const char url *` there, it would be hidden on an extended one of reference backend. Is that right? If so, how to do this flow on flow?
Nowadays, `update/check` outputs the JSON (soon-to-be composed packet) with information (including URL, that should not be there) to feed `update/fetch`, which in turn outputs the `file_path` to `update/install`. What could be sent in those cases from one node to another? A custom packet? Can they contain the extensible structs `sol_update_info` and `sol_update`?
About them,  I'd rather `sol_update_fetch_info` and `sol_update_install_info` - easy on C API, but not sure about on flow...